### ISSUE TITLE
`SetSystemTrayIcon` does not actually work until runtime.

### ISSUE THREAD

**Issue by flowchartsman** [fyne-io/fyne#4548]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

If you create an application that uses a system tray by calling `SetSystemTrayMenu`, there is no way to set an initial system tray icon that is different to the default app Icon. This is an annoyance on MacOS, where colored icons look out of place (and are in fact discouraged in the menu bar in favor of [template images](https://developer.apple.com/documentation/appkit/nsimage/1520017-template)[^1], which will do the right thing in light/dark/high-contrast mode. There is no way to alter this behavior, leading to an app that looks kind of unprofessional as it creates a colored menubar icon, only to quickly switch it out for a different one after a startup delay.

[^1]: This is supported in `fyne.io/systray` by calling `SetTemplateIcon`, however `fyne.io/v2/app` does not support this, and it probably should. Will file another issue.

### How to reproduce

Run the below code. The application will always display the initial application icon in the system tray before beginning to cycle through the others.

### Example code

```go
package main

import (
	"time"

	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/driver/desktop"
)

//go:generate fyne bundle -o systray_icons.go icon/tb0.png
//go:generate fyne bundle -o systray_icons.go -append icon/tb1.png
//go:generate fyne bundle -o systray_icons.go -append icon/tb2.png
//go:generate fyne bundle -o systray_icons.go -append icon/tb3.png
//go:generate fyne bundle -o systray_icons.go -append icon/tb4.png
//go:generate fyne bundle -o systray_icons.go -append icon/tbalert.png
func main() {
	a := app.New()
	if app, ok := a.(desktop.App); ok {
		m := &fyne.Menu{Label: "App"}
		app.SetSystemTrayMenu(m)
		app.SetSystemTrayIcon(resourceTb4Png) // does not work
		iconcycle := []fyne.Resource{
			resourceTb3Png,
			resourceTb2Png,
			resourceTb1Png,
			resourceTb0Png,
			resourceTbalertPng,
			resourceTb4Png,
		}
		go func() {
			i := 0
			for range time.NewTicker(time.Second).C {
				app.SetSystemTrayIcon(iconcycle[i])
				i++
				if i == len(iconcycle) {
					i = 0
				}
			}
		}()
	}
	a.Lifecycle().SetOnStarted(func() {
		a.(desktop.App).SetSystemTrayIcon(resourceTb4Png) // also does not work
	})
	a.Run()
}
```

### Fyne version

this bug is present in HEAD

### Go compiler version

irrelevant

### Operating system and version

macos 14.1.1

### Additional Information

The reason this happens is that `*gLDriver.SetSystemTrayMenu` [calls `fyne.io/systray.RunWithExternalLoop`](https://github.com/fyne-io/fyne/blob/master/internal/driver/glfw/driver_desktop.go#L40), which consumes a callback to be run when the systray loop is [started later](https://github.com/fyne-io/fyne/blob/master/internal/driver/glfw/loop.go#L123-L125). This callback sets the default icon, overwriting anything the user might have attempted to set in the interim.

There are a few different ways to solve this issue; however, the deferred startup makes it difficult. The method with the smallest blast radius is probably to alter `SetSystemTrayIcon` to write an atomic or mutex-guarded value to the app struct representing the current system tray icon just before the call to set it. Then the callback could reference this value when starting the systray loop. This would allow the user to set the icon early.

**Comment by andydotxyz** [comment]
I cannot replicate this issue. The following code will set an icon that is not the same as the app icon:

```go
	if desk, ok := a.(desktop.App); ok {
		desk.SetSystemTrayIcon(theme.FileIcon())
		desk.SetSystemTrayMenu(fyne.NewMenu("Empty"))
	}
```

**Comment by flowchartsman** [comment]
Ah, it seems I was partially mistaken. I'm not sure why it wasn't working when using the app call before, but I've since upgraded and now the reason is that I am calling `fyne.io/systray.SetTemplateIcon` directly, and it doesn't have access to the [package-level variable](https://github.com/fyne-io/fyne/blob/master/internal/driver/glfw/driver_desktop.go#L160) that gets [picked up later](https://github.com/fyne-io/fyne/blame/master/internal/driver/glfw/driver_desktop.go#L41). I guess a fix for #4549 will probably fix that.

Closing.
